entrypoint.sh

```diff
@@ -26,18 +26,16 @@ test_vars() {
 
 aws() {
 test_vars AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY
-set -x
 
-if [[ "${PODVM_LAUNCHTEMPLATE_NAME}" ]]; then
-	optionals+="-use-lt -aws-lt-name ${PODVM_LAUNCHTEMPLATE_NAME} "
-else
-	optionals+="-imageid ${PODVM_AMI_ID} "
-	optionals+="-instance-type ${PODVM_INSTANCE_TYPE:-t3.small} "
-	optionals+="-securitygroupid ${AWS_SG_ID} "
-	optionals+="-keyname ${SSH_KP_NAME} "
-	optionals+="-subnetid ${AWS_SUBNET_ID} "
-fi
+[[ "${PODVM_LAUNCHTEMPLATE_NAME}" ]] && optionals+="-use-lt -aws-lt-name ${PODVM_LAUNCHTEMPLATE_NAME} " # has precedence if set
+[[ "${AWS_SG_ID}" ]] && optionals+="-securitygroupid ${AWS_SG_ID} " # MUST if template is not used
+[[ "${PODVM_AMI_ID}" ]] && optionals+="-imageid ${PODVM_AMI_ID} " # MUST if template is not used
+[[ "${PODVM_INSTANCE_TYPE}" ]] && optionals+="-instance-type ${PODVM_INSTANCE_TYPE} " # default t3.small
+[[ "${SSH_KP_NAME}" ]] && optionals+="-keyname ${SSH_KP_NAME} " # if not retrieved from IMDS
+[[ "${AWS_SUBNET_ID}" ]] && optionals+="-subnetid ${AWS_SUBNET_ID} " # if not set retrieved from IMDS
+[[ "${AWS_REGION}" ]] && optionals+="-aws-region ${AWS_REGION} " # if not set retrieved from IMDS
```

Comment:
hmm doesn't the above changes the logic? Earlier the ami, instance etc were only needed when launch template was not used. Now it's not the case..

Reply:
yes, variables passing logic has changed but caa logic is the same,  i.e. if you choose to set these ami, subnet.. vars in addition to launch template they will be passed to caa but will be ignored in favor of using the template.
I can change it but i think it makes more sense to avoid doing non-mandatory logic (which is already done in the caa code) in the entrypoint.sh, additionally, it will allow to have template fallback if we wish (in the future)

Reply:
I see.. Thanks for the explanation.